### PR TITLE
Cave Story: Revise Sand Zone

### PR DESCRIPTION
**Pull request by Miepee**
Health numbers for convenience:

Curly:
14 -> 21
9 -> 14

Omega:
19 -> 29
10 -> 15

Toroko+
22 -> 33
15 -> 23
9 -> 14

### CODE REVIEW

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 1]
```diff
@@ -120,38 +131,61 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
           Any of the following:
+              # Break the Blocks
               Can Break Blocks
+              # Break Blocks with Bubbler
               Bubbler and Break blocks with Bubbler (Beginner)
-              Missiles ≥ 4 and Break blocks with Missiles (Intermediate)
-              Missiles ≥ 2 and Break blocks with Missiles (Advanced)
-              Missiles and Break blocks with Missiles (Expert)
+              All of the following:
+                  # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:172]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 2]
```diff
@@ -120,38 +131,61 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+                  Any of the following:
+                      Missiles ≥ 4 and Break blocks with Missiles (Intermediate)
+                      Missiles ≥ 2 and Break blocks with Missiles (Advanced)
+                      Missiles and Break blocks with Missiles (Expert)
   > Pawprint Spot
       All of the following:
           After Defeated Omega
           Any of the following:
               Can Break Blocks
+              # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:183]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 3]
```diff
@@ -120,38 +131,61 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
               Bubbler and Break blocks with Bubbler (Beginner)
-              Missile Launcher ≥ 3 and Break blocks with Missiles (Intermediate)
+              All of the following:
+                  # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:186]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 4]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
           Any of the following:
               # Enter Arena
               Can Break Blocks
+              # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:199]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 5]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
               Bubbler and Break blocks with Bubbler (Beginner)
-              Missiles ≥ 6 and Break blocks with Missiles (Intermediate)
-              Missiles ≥ 2 and Break blocks with Missiles (Advanced)
-              Missiles and Break blocks with Missiles (Expert)
+              All of the following:
+                  # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:202]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 6]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
-              Can Kill Bosses
-              Missiles ≥ 11 and Super Missile Launcher and Kill Bosses with Missiles (Beginner)
-              Missiles ≥ 10 and Super Missile Launcher and Kill Bosses with Missiles (Intermediate)
-              Missiles ≥ 50 and Missile Launcher and Kill Bosses with Missiles (Advanced)
-              Missiles ≥ 25 and Missile Launcher and Kill Bosses with Missiles (Hypermode)
+              Any of the following:
+                  # Item requirements
+                  Can Kill Bosses
+                  All of the following:
+                      # Kill with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:214]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 7]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+                      Missile Launcher
+                      Any of the following:
+                          Missiles ≥ 50 and Kill Bosses with Missiles (Advanced)
+                          Missiles ≥ 25 and Kill Bosses with Missiles (Hypermode)
+                  All of the following:
+                      # Kill with Super Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:220]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 8]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+                          Missiles ≥ 11 and Kill Bosses with Missiles (Beginner)
+                          Missiles ≥ 10 and Kill Bosses with Missiles (Intermediate)
+              Any of the following:
+                  # Energy Requirements
+                  Combat (Intermediate) or Normal Damage ≥ 19
+                  Combat (Beginner) and Normal Damage ≥ 10
   > Above Sunstones
       Any of the following:
           Can Break Blocks
+          # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:232]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 9]
```diff
@@ -162,24 +196,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
           Bubbler and Break blocks with Bubbler (Beginner)
-          Missiles ≥ 22 and Break blocks with Missiles (Intermediate)
-          Missiles ≥ 11 and Break blocks with Missiles (Advanced)
-          Missiles and Break blocks with Missiles (Expert)
+          All of the following:
+              # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:235]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 10]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
   > Lower Side
       Any of the following:
           Can Break Blocks
+          # Break Blocks with Bubbler
           Bubbler and Break blocks with Bubbler (Beginner)
-          Missiles ≥ 3 and Break blocks with Missiles (Intermediate)
-          Missiles and Break blocks with Missiles (Advanced)
+          All of the following:
+              # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:254]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 11]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
   > Lower Side
       Any of the following:
           Can Break Blocks
+          # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:251]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 12]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+              # Break Blocks with Missiles
+              Missile Launcher or Super Missile Launcher
+              Any of the following:
+                  Missiles ≥ 3 and Break blocks with Missiles (Intermediate)
+                  Missiles and Break blocks with Missiles (Advanced)
   > Above Sunstones
       All of the following:
           After Defeated Omega
           Any of the following:
+              # Break the Blocks
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:263]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 13]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
               Can Break Blocks
-              Missiles ≥ 4 and Break blocks with Missiles (Intermediate)
-              Missiles ≥ 2 and Break blocks with Missiles (Advanced)
-              Missiles and Break blocks with Missiles (Expert)
+              # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:265]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 14]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
               Bubbler and Break blocks with Bubbler (Beginner)
+              All of the following:
+                  # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:268]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 15]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+                  Missile Launcher or Super Missile Launcher
+                  Any of the following:
+                      Missiles ≥ 4 and Break blocks with Missiles (Intermediate)
+                      Missiles ≥ 2 and Break blocks with Missiles (Advanced)
+                      Missiles and Break blocks with Missiles (Expert)
   > Pawprint Spot
       Any of the following:
           Can Break Blocks
+          # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:277]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 16]
```diff
@@ -190,23 +248,38 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
           Bubbler and Break blocks with Bubbler (Beginner)
-          Missile Launcher ≥ 3 and Break blocks with Missiles (Intermediate)
+          All of the following:
+              # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:280]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 17]
```diff
@@ -219,13 +292,21 @@ Extra - starting_script: <TRA0010:0099:0036:0033
           After Defeated Omega
           Any of the following:
               Can Break Blocks
+              # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:295]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 18]
```diff
@@ -219,13 +292,21 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
               Bubbler and Break blocks with Bubbler (Beginner)
-              Missile Launcher ≥ 3 and Break blocks with Missiles (Intermediate)
+              All of the following:
+                  # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:298]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 19]
```diff
@@ -219,13 +292,21 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+                  Missiles ≥ 3 and Break blocks with Missiles (Intermediate)
+                  Missile Launcher or Super Missile Launcher
   > Outside Jenka's House
       Any of the following:
           Can Break Blocks
+          # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:304]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 20]
```diff
@@ -219,13 +292,21 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
           Bubbler and Break blocks with Bubbler (Beginner)
-          Missile Launcher ≥ 3 and Break blocks with Missiles (Intermediate)
+          All of the following:
+              # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:307]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 21]
```diff
@@ -258,7 +339,14 @@ Extra - starting_script: <TRA0010:0099:0036:0033
   > Exit to Sand Zone Storehouse
       After Returned Puppies
   > Refill (Lower)
-      Can Break Blocks
+      Any of the following:
+          Can Break Blocks
+          # Break Blocks with Bubbler
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:344]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 22]
```diff
@@ -258,7 +339,14 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
+          Bubbler and Break blocks with Bubbler (Beginner)
+          All of the following:
+              # Break Blocks with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:347]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 23]
```diff
@@ -324,9 +412,16 @@ Extra - starting_script: <TRA0029:0090:0018:0009
...
   * Layers: default
   > Event - Curly
-      Any of the following:
-          Can Kill Bosses
-          Missiles ≥ 9 and Super Missile Launcher and Kill Bosses with Missiles (Beginner)
+      All of the following:
+          Any of the following:
+              # Item requirements
+              Can Kill Bosses
+              # Kill with Super Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:419]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 24]
```diff
@@ -465,10 +560,19 @@ Extra - map_name: Gard
...
   > Event - Toroko+
-      Any of the following:
-          Can Kill Bosses
-          Missiles ≥ 13 and Super Missile Launcher and Kill Bosses with Missiles (Intermediate)
-          Missiles ≥ 30 and Missile Launcher and Kill Bosses with Missiles (Expert)
+      All of the following:
+          Any of the following:
+              # Item requirements
+              Can Kill Bosses
+              # Kill with Super Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:567]: .

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 25]
```diff
@@ -465,10 +560,19 @@ Extra - map_name: Gard
...
+              Missiles ≥ 13 and Super Missile Launcher and Kill Bosses with Missiles (Intermediate)
+              # Kill with Missiles
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:569]: .

**Review by duncathan — CHANGES_REQUESTED** [review]

**Comment by Miepee** [comment]
Removed all those comments except for the Omega fight due to the omega requirements being a bit big.

File: randovania/games/cave_story/logic_database/Sand Zone.txt [hunk 26]
```diff
@@ -121,37 +125,48 @@ Extra - starting_script: <TRA0010:0099:0036:0033
...
           Bubbler and Break blocks with Bubbler (Beginner)
-          Missiles ≥ 27 and Break blocks with Missiles (Intermediate)
-          Missiles ≥ 13 and Break blocks with Missiles (Advanced)
-          Missiles and Break blocks with Missiles (Expert)
+          All of the following:
+              Missile Launcher or Super Missile Launcher
+              Any of the following:
+                  Missiles ≥ 27 and Break blocks with Missiles (Intermediate)
+                  Missiles ≥ 13 and Break blocks with Missiles (Advanced)
+                  Missiles and Break blocks with Missiles (Expert) and Can Kill Bosses
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Sand Zone.txt:148]: this specific one is worth a comment mentioning that you're farming missile drops

**Review by duncathan — CHANGES_REQUESTED** [review]

**Review by duncathan — APPROVED** [review]

**Pull request merged**